shared/src/api/extension/flatExtensionApi.ts

```diff
@@ -40,10 +40,19 @@ export type PartialWorkspaceNamespace = Omit<
  *
  * @param mainAPI
  */
-export const initNewExtensionAPI = (mainAPI: Remote<MainThreadAPI>): InitResult => {
-    const state: ExtState = { roots: [], versionContext: undefined }
+export const initNewExtensionAPI = (
+    mainAPI: Remote<MainThreadAPI>,
+    initialSettings: Readonly<SettingsCascade<object>>
+): InitResult => {
+    const state: ExtState = { roots: [], versionContext: undefined, settings: initialSettings }
 
     const configChanges = new ReplaySubject<void>(1)
```

Comment:
Following the discussion below, using `BehaviorSubject` directly and removing the `configChanges.next()` makes more sense (otherwise we're manually "molding" a `ReplaySubject` into a `BehaviorSubject`).
```suggestion
    const configChanges = new BehaviorSubject<void>(undefined)
```